### PR TITLE
fixes for legacy content on some sites

### PR DESCRIPTION
**Pull request by timatron**

### CODE REVIEW

File: php/class-photonfill-transform.php [hunk 1]
```diff
@@ -111,14 +111,26 @@ public function transform_photon_url( $args, $image_url, $scheme = null ) {
 		 * Get the width and height of the default transform. Can be used for other transforms.
```
Comment (mboynes) [inline php/class-photonfill-transform.php]: Add docs for `$args`

File: php/class-photonfill-transform.php [hunk 2]
```diff
@@ -111,14 +111,26 @@ public function transform_photon_url( $args, $image_url, $scheme = null ) {
...
 		 */
 		public function get_dimensions( $args ) {
+			$width = ( ! empty( $args['width'] ) ) ? $args['width'] : 1000;
+			$height = ( ! empty( $args['height'] ) ) ? $args['height'] : 1000;
```
Comment (mboynes) [inline php/class-photonfill-transform.php]: * Why is this necessary?
* Why 1000 for these?
* Under what circumstances would they be empty, and are there any valid cases that this would then break?

File: php/class-photonfill-transform.php [hunk 3]
```diff
@@ -111,14 +111,26 @@ public function transform_photon_url( $args, $image_url, $scheme = null ) {
...
+
+			// An extra way to get that data.
+			if ( empty( $args['width'] ) && empty( $args['height'] ) && ! empty( $args['resize'] ) ) {
+				$dims = explode( ',', $args['resize'] );
+				if ( ! empty( $dims[0] ) && ! empty( $dims[1] ) ) {
+					$width = $dims[0];
+					$height = $dims[1];
```
Comment (mboynes) [inline php/class-photonfill-transform.php]: neat trick:

```php
list( $width, $height ) = $dims;
```

File: php/class-photonfill-transform.php [hunk 4]
```diff
@@ -111,14 +111,26 @@ public function transform_photon_url( $args, $image_url, $scheme = null ) {
...
+				}
+			}
+
 			// We are only going to use the size if none are defined in the transform, which shouldn't happen.
 			if ( isset( $args['crop'] ) && false === $args['crop'] ) {
 				$h = 100;
 			} else {
-				$h = $args['height'] . 'px';
+				$h = $height . 'px';
```
Comment (mboynes) [inline php/class-photonfill-transform.php]: This is really confusing -- why the `px`? I know that predates this PR, but no time like the present to fix it. To clarify, there's a lot of inconsistency with what this function will return. width may be an int or it may be a numeric string (or theoretically anything else, since validation isn't done on it, but we'll make that assumption). height may be an int, a numeric string, or a numeric string with 'px' at the end. maybe these are all valid, but that needs to be clearly documented.

File: php/class-photonfill-transform.php [hunk 5]
```diff
@@ -111,14 +111,26 @@ public function transform_photon_url( $args, $image_url, $scheme = null ) {
 		 * Get the width and height of the default transform. Can be used for other transforms.
 		 */
 		public function get_dimensions( $args ) {
+			$width = ( ! empty( $args['width'] ) ) ? $args['width'] : 1000;
+			$height = ( ! empty( $args['height'] ) ) ? $args['height'] : 1000;
+
+			// An extra way to get that data.
+			if ( empty( $args['width'] ) && empty( $args['height'] ) && ! empty( $args['resize'] ) ) {
+				$dims = explode( ',', $args['resize'] );
```
Comment (mboynes) [inline php/class-photonfill-transform.php]: in line with my last note, this is going to lead to `$width` and `$height` being strings

File: php/class-photonfill-transform.php [hunk 6]
```diff
@@ -111,14 +111,26 @@ public function transform_photon_url( $args, $image_url, $scheme = null ) {
...
+				if ( ! empty( $dims[0] ) && ! empty( $dims[1] ) ) {
+					$width = $dims[0];
+					$height = $dims[1];
+				}
+			}
+
 			// We are only going to use the size if none are defined in the transform, which shouldn't happen.
 			if ( isset( $args['crop'] ) && false === $args['crop'] ) {
 				$h = 100;
```
Comment (mboynes) [inline php/class-photonfill-transform.php]: if `$args['crop'] === false`, all the work you've done around `$height` was pointless

**Comment by willgladstone** [comment]
Gonna close this as I don't think these are relevant anymore.
